test/Test/Database/LSMTree/ModelIO/Normal.hs

```diff
@@ -48,6 +51,33 @@ makeNewTable h ups = do
     updates hdl ups
     return (s, hdl)
 
+-- | Like 'retrieveBlobs' but works for any 'Traversable'.
+-- 
+-- Like 'partsOf' in @lens@ this uses state monad.
+retrieveBlobsTrav ::
+  (IsTableHandle h, IOLike m, SomeSerialisationConstraint blob, Traversable t)
+  => h m k v blob -> t (BlobRef h blob) -> m (t blob)
+retrieveBlobsTrav hdl brefs = do
+  blobs <- retrieveBlobs hdl (toList brefs)
+  evalStateT (traverse (\_ -> state un) brefs) blobs
```

Comment:
it took me a while to figure out what this is doing

Reply:
... but it says it in the comment.

Reply:
longer explanation https://oleg.fi/gists/posts/2023-10-12-use-traversals-for-batch-operations.html